src/weave/smoother.py

```diff
@@ -249,77 +250,72 @@ def check_args(data: DataFrame, observed: List[str],
         TypeError
             If `smoother` arguments contain invalid types.
         ValueError
-            If either `observed` or `smoothed` is an empty list or
-            contain duplicates.
+            If `observed`, `stdev`, or `smoothed` overlap.
 
         """
         # Check types
         if not isinstance(data, DataFrame):
             raise TypeError('`data` is not a DataFrame')
-        if not all(isinstance(col, str) for col in observed):
-            raise TypeError('`observed` contains invalid types')
-        if smoothed is not None:
-            if not all(isinstance(col, str) for col in smoothed):
-                raise TypeError('`smoothed` contains invalid types')
+        if not isinstance(observed, str):
+            raise TypeError('`observed` is not a str')
+        if stdev is not None and not isinstance(stdev, str):
+            raise TypeError('`stdev` is not a str')
+        if smoothed is not None and not isinstance(smoothed, str):
+            raise TypeError('`smoothed` is not a str')
         if fit is not None and not isinstance(fit, str):
             raise TypeError('`fit` is not a str')
         if predict is not None and not isinstance(predict, str):
             raise TypeError('`predict` is not a str')
 
         # Check values
-        if len(observed) == 0:
-            raise ValueError('`observed` is an empty list')
-        if len(observed) > len(set(observed)):
-            raise ValueError('`observed` contains duplicates')
-        if smoothed is not None:
-            if len(smoothed) == 0:
-                raise ValueError('`smoothed` is an empty list')
-            if len(smoothed) > len(set(smoothed)):
-                raise ValueError('`smoothed` contains duplicates')
-            if len(observed) != len(smoothed):
-                msg = 'Length of `observed` and `smoothed` do not match'
-                raise ValueError(msg)
-            if len(set(observed).intersection(set(smoothed))) > 0:
-                raise ValueError('Duplicates in `observed` and `smoothed`')
-
-    def check_data(self, data: DataFrame, observed: List[str],
-                   smoothed: List[str], fit: Optional[str],
+        if stdev is not None:
+            if observed == stdev:
+                raise ValueError('`observed` == `stdev`')
+            if smoothed is not None and stdev == smoothed:
+                raise ValueError('`stdev` == `smoothed`')
+        if smoothed is not None and observed == smoothed:
+            raise ValueError('`observed` == `smoothed`')
```

Comment:
Here maybe we can use a `set` to see if there are any overlap between `observed, stdev, smoothed`

Reply:
Good idea